### PR TITLE
chore(ansible): rework testing and linting workflows

### PR DESCRIPTION
**Pull request by CherryKitten**

### CODE REVIEW

**Comment by CherryKitten** [comment]
This is finally ready for review, you can see the updated workflows running on all our ansible collections in the PRs that are linked above.

The only fails left are because of failing lints, all errors and github weirdnesses that I ran into should be fixed

Some playbooks still fail the syntax-check lint because they can't find certain collections they depend on, but it seems like this was already a problem on GitLab and I'd move fixing that to a later follow-up.

File: .github/workflows/ansible.yml [hunk 1]
```diff
@@ -7,24 +7,38 @@ on:
...
+    name: "Run integration tests"
+    uses: famedly/github-workflows/.github/workflows/ansible-test.yml@sammy/ansible-improvements
     with:
-      collection: ${{ inputs.collection }}
       testing-type: integration
+      collection: ${{ inputs.collection }}
   black:
-    uses: famedly/github-workflows/.github/workflows/black.yml
+    name: "Format using black"
+    uses: famedly/github-workflows/.github/workflows/black.yml@sammy/ansible-improvements
```
Comment (jcgruenhage) [inline .github/workflows/ansible.yml:42]: This, and every other occurance of it, needs to be fixed

**Review by jcgruenhage — CHANGES_REQUESTED** [review]
lgtm otherwise

File: .github/workflows/ansible.yml (hunk 1 above)
Comment (CherryKitten) [reply .github/workflows/ansible.yml]: whoops, fixed

**Review by jcgruenhage — APPROVED** [review]

**Pull request merged**